src/pages/experiments/[experimentId]/data-processing/index.jsx

```diff
@@ -66,17 +66,34 @@ const DataProcessingPage = ({ experimentId, experimentData, route }) => {
   ];
 
   const dispatch = useDispatch();
-  const [stepIdx, setStepIdx] = useState(0);
 
   const completedPath = '/experiments/[experimentId]/data-exploration';
 
-  const completedSteps = useSelector((state) => state.experimentSettings.processing.meta.stepsDone);
+  const completedSteps = useSelector((state) => state.experimentSettings.processing.processingConfig.stepsDone);
+
+  const [stepIdx, setStepIdx] = useState((completedSteps?.size ?? 0) % steps.length);
+
   const carouselRef = useRef(null);
 
   useEffect(() => {
-    if (carouselRef.current) {
-      carouselRef.current.goTo(stepIdx);
-    }
+    dispatch(loadProcessingSettings(experimentId));
+  }, [experimentId]);
+
+  useEffect(() => {
+    const goToStepIdx = () => {
+      if (carouselRef.current) {
+        carouselRef.current.goTo(stepIdx);
+      }
+    };
+
+    const completeProcessingStepIfAdvanced = () => {
+      if (stepIdx > completedSteps.size) {
+        dispatch(completeProcessingStep(experimentId, steps[stepIdx - 1].key, steps.length));
+      }
+    };
+
+    goToStepIdx();
+    completeProcessingStepIfAdvanced();
```

Comment:
Any reason why `goToStepIdx()` and `completeProcessingStepIfAdvanced()` are implemented as functions instead of just blocks of code?

Reply:
just for readability, and differentiation of responsibilities of each of the blocks of code, we are doing two completely different things that are based on the value that `stepIdx` takes

Reply:
Hmm... the body of the functions are quite simple (functions wrapped in if statements). I don't think you need to put them in functions... If you want to explain what they do, I think you can put a line of comment on top of each of the function,

Reply:
I think I did it also to show the difference in functionality between both of them, is there any issue with having them as auxiliary functions inside the useEffect they are being used in?

I also would prefer to avoid comments when it is possible to have it expressed in the code

Reply:
no, not really a problem